File: ProgramacionMultimediaDispositivosMoviles/PMDM07/PMDM_Tarea03/PMDM03_Tarea_Moga/app/src/main/java/com/example/pmdm03_tarea_moga/MainActivity.java
package com.example.pmdm03_tarea_moga;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import androidx.core.content.ContextCompat;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.android.material.switchmaterial.SwitchMaterial;

// Clase principal de la aplicación que maneja la pantalla de login
public class MainActivity extends AppCompatActivity {

    // Declaración de variables para los elementos de la UI
    private Button botonInstrucciones;
    private Button botonComenzar;
    private EditText campoNombre;
    private SwitchMaterial switchModo;

    // Constantes para almacenar/recuperar preferencias del usuario SharedPreferences
    private static final String PREFS_NAME = "MisPreferencias"; // Nombre del archivo
    private static final String MODO_KEY = "modo_practica";     // Clave para guardar





    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);

        // Establecer el layout de la pantalla de login
        setContentView(R.layout.pagina_login);

        // INICIALIZACIÓN DE VISTAS
        // Conectar las variables Java con los elementos del XML
        campoNombre = findViewById(R.id.campoNombre);
        botonComenzar = findViewById(R.id.comenzar);
        botonInstrucciones = findViewById(R.id.instrucciones);
        switchModo = findViewById(R.id.switchModo);


        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.login), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });




        // GESTIÓN DE PREFERENCIAS (SHAREDPREFERENCES)
        // Obtener acceso al almacenamiento persistente de preferencias
        SharedPreferences prefs = getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);

        //  Leer el valor guardado. Si no existe nada, ponemos 'false' (Modo Evaluación)
        boolean modoPracticaActivado = prefs.getBoolean(MODO_KEY, false);

        // Aplicar el valor al switch y el texto correspondiente
        switchModo.setChecked(modoPracticaActivado);
        if (modoPracticaActivado) {
            switchModo.setText("Modo Práctica");  // Texto cuando está activado
        } else {
            switchModo.setText("Modo Evaluación");  // Texto cuando está desactivado
        }

        // Guardar cambios cuando el usuario mueve el switch
        switchModo.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                // Actualizar texto del switch segun su estado
                if (isChecked) {
                    switchModo.setText("Modo Práctica");
                } else {
                    switchModo.setText("Modo Evaluación");
                }

                // Guardar el nuevo estado en SharedPreferences para persistencia
                SharedPreferences.Editor editor = prefs.edit();
                editor.putBoolean(MODO_KEY, isChecked);
                editor.apply();  // Guardar cambios de forma asíncrona
            }
        });

        //  LÓGICA BOTÓN COMENZAR
        botonComenzar.setOnClickListener(v -> {
            // Validar que el usuario haya ingresado un nombre
            if (validarUsuario()) {
                // Crear intent para ir a la actividad de preguntas
                Intent intent = new Intent(MainActivity.this, PreguntasActivity.class);
                enviarDatosExtra(intent);
                startActivity(intent);
            }
        });


        //  LÓGICA BOTÓN INSTRUCCIONES
        botonInstrucciones.setOnClickListener(v -> {
            // Validar que el usuario haya ingresado un nombre
            if (validarUsuario()) {
                // Crear intent para ir a la actividad de instrucciones
                Intent intent = new Intent(MainActivity.this, InstruccionesActivity.class);
                enviarDatosExtra(intent);
                startActivity(intent);
            }
        });
    }

    // Metodo para empaquetar los datos que se enviarán a otras actividades
    private void enviarDatosExtra(Intent intent) {
        intent.putExtra("nombre_usuario", campoNombre.getText().toString().trim());
        intent.putExtra("modo_practica", switchModo.isChecked());
    }

    // Metodo para validar que el usuario haya ingresado un nombre
    private boolean validarUsuario() {
        String nombreUsuario = campoNombre.getText().toString().trim();
        if (nombreUsuario.isEmpty()) {
            Toast.makeText(this, "Debes introducir un nombre de usuario", Toast.LENGTH_SHORT).show();
            return false;
        }
        return true;
    }

}

File: DesarrolloInterfaces/DI03/Juego_botones/app/src/main/java/com/example/juego_botones/MainActivity.kt
package com.example.juego_botones

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.LinearLayout
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import kotlin.random.Random

class MainActivity : AppCompatActivity() {

    private val numBotones = 10
    private lateinit var llBotonera: LinearLayout

    // Variable para guardar el numero del boton ganador
    private var botonGanador = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        llBotonera = findViewById(R.id.llBotonera)

        // Se genera un numero aleatorio entre 0 y numBotones
        botonGanador = Random.nextInt(numBotones)


        // Se crean las propiedades de layout que tendran los botones
        val lp = LinearLayout.LayoutParams(
            LinearLayout.LayoutParams.MATCH_PARENT,
            resources.getDimensionPixelSize(R.dimen.button_height) // Se define la altura del boton con un valor concreto
        )

        // Se crean los botones en bucle
        for (i in 0 until numBotones){
            val button = Button(this)

            // Se asignan las propiedades de layout al boton
            button.layoutParams = lp

            // Se asigna el texto al boton
            button.text = "Botón " + String.format("%02d", i)

            // Se asigna el Listener
            button.setOnClickListener{

                // Se verifica si el boton pulsado es el ganador
                if(i == botonGanador){
                    Toast.makeText(this,"ME ENCONTRASTE!!", Toast.LENGTH_SHORT).show()
                } else {
                    Toast.makeText(this,"Sigue buscando", Toast.LENGTH_SHORT).show()
                }
            }

            llBotonera.addView(button)
        }



        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    // Funcion privada para mostrar mensaje al pulsar un boton
    private fun buttonClickListener(index: Int) : View.OnClickListener{
        return View.OnClickListener{
            Toast.makeText(this@MainActivity,
                "Botón " + String.format("%02d", index) + " pulsado",
                Toast.LENGTH_SHORT).show()
        }
    }
}